src/download.test.js

```diff
@@ -55,3 +62,29 @@ describe("open", () => {
     });
   });
 });
+
+describe("openHns", () => {
+  const hnsUrl = `${portalUrl}/hns/${hnsLink}`;
+
+  beforeEach(() => {
+    mock.onGet(hnsUrl).reply(200, { skylink: skylink });
+  });
+
+  it("should call axios.get with the portal and hns link", async () => {
+    const windowOpen = jest.spyOn(window, "open").mockImplementation();
+
+    let i = 1;
+    for (const input of validHnsLinkVariations) {
+      await client.openHns(input);
+
+      expect(mock.history.get.length).toBe(i);
+
+      expect(windowOpen).toHaveBeenCalledTimes(i);
+      expect(windowOpen).toHaveBeenCalledWith(`${portalUrl}/${skylink}`, "_blank");
+
+      i++;
+    }
+
+    mock.resetHistory();
```

Comment:
It's better to always reset mock before the assertions because in case of the failure, the reset might not be executed and subsequent tests will report fake success/failures. It might not be a case here but it's a good practice. Also it's a common practice in js to use function for looping array instead of an actual loop for increased safety of leaking variables out - again, not a case here but a good practice.

```suggestion
    validHnsLinkVariations.forEach((input, index) => {
      mock.resetHistory();
      
      await client.openHns(input);

      expect(mock.history.get.length).toBe(1);

      expect(windowOpen).toHaveBeenCalledTimes(1);
      expect(windowOpen).toHaveBeenCalledWith(`${portalUrl}/${skylink}`, "_blank");
    }
```

Reply:
`forEach` doesn't work if you pass it an `async` function, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach